client/src/components/TopBar.tsx

```diff
@@ -28,11 +30,32 @@ const TopBar = ({
   toggleMenuAction,
   handleSecurityBannerBottom
 }: TopBarProps) => {
-  const { appSettings, currentUser } = useContext(AppContext)
+  const { appSettings, currentUser, loadAppData } = useContext(AppContext)
   const [bannerVisibility, setBannerVisibility] = useState(false)
   const [height, setHeight] = useState(0)
   const topbarDiv = useRef()
 
+  useEffect(() => {
+    let isLoading = false
+
+    const tick = async () => {
+      if (isLoading) {
+        return
+      }
+      isLoading = true
+      try {
+        await loadAppData()
```

Comment:
`loadAppData()` is quite a heavy query if you're only interested in are the new adminSettings. Perhaps use a similar approach to `connectionUtils.ts` and periodically only query the adminSettings?